### PR TITLE
Convert encode/decode of DSS signatures to Rust

### PR DESCRIPTION
**Pull request by alex**
Before this merges, I'd like to benchmark this to make sure we're going zoom zoom.

### CODE REVIEW

**Comment by alex** [comment]
Benchmark:

```
python -m timeit -s "from cryptography.hazmat.primitives.asymmetric.utils import decode_dss_signature" "decode_dss_signature(b'0-\x02\x15\x00\xb5\xaf0xg\xfb\x8bT9\x00\x13\xccg\x02\r\xdf\x1f,\x0b\x81\x02\x14b\r;\"\xabP1D\x0c>5\xea\xb6\xf4\x81)\x8f\x9e\x9f\x08')"
```

Before: 6.4 usec per loop
After: 730 nsec per loop

Seems good

**Comment by alex** [comment]
Benchmark:

```
(.venv) alex@media-01:/tmp/x$ python -m timeit -s "from cryptography.hazmat.primitives.asymmetric.utils import encode_dss_signature" "encode_dss_signature(21312412321, 32143214321)"
```

Before: 2.6 usec per loop
After: 1 usec per loop

Seems good

File: src/rust/src/asn1.rs [hunk 1]
```diff
@@ -95,11 +96,71 @@ fn parse_spki_for_data(py: pyo3::Python<'_>, data: &[u8]) -> pyo3::PyResult<pyo3
...
+    v.call_method1("to_bytes", (n, "big"))?.extract()
+}
+
+#[pyo3::prelude::pyfunction]
+fn encode_dss_signature(
+    py: pyo3::Python<'_>,
+    r: &pyo3::types::PyLong,
+    s: &pyo3::types::PyLong,
+) -> pyo3::PyResult<pyo3::PyObject> {
+    if r.rich_compare((0).to_object(py), CompareOp::Lt)?
```
Comment (reaperhulk) [inline src/rust/src/asn1.rs:134]: How do we ensure coverage when hoisting these types of branches into rust?

Comment (alex) [reply src/rust/src/asn1.rs:134]: Well, we could figure out Rust coverage in CI? Do you want that to be a pre-req to this? (Is this literally the first branch in the rust code? Neat!)

Comment (reaperhulk) [reply src/rust/src/asn1.rs:134]: `parse_spki_for_data` has a branch too, but it slipped my mind in that review 😄

Yes I think we should figure that out, although I'm a bit worried given that our hosted coverage service doesn't handle a single language well...

Comment (alex) [reply src/rust/src/asn1.rs:134]: Yeah, tooling here will be... fascinating. I'll investigate.

**Comment by reaperhulk** [comment]
This should be rebase-able now 😄

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**